dashboard/test/models/concerns/user/verifiable_test.rb

```diff
@@ -0,0 +1,117 @@
+require 'test_helper'
+
+class VerifiableTest < ActiveSupport::TestCase
+  include Minitest::RSpecMocks
+
+  describe '#verify_teacher!' do
+    let(:teacher) {create :teacher}
+
+    before do
+      UserPermission.any_instance.stubs(:send_verified_teacher_email)
+      teacher.verify_teacher!
+    end
+
+    it 'sets the permission correctly' do
+      permission_values = teacher.permissions.map(&:permission)
+      _(permission_values).must_include UserPermission::AUTHORIZED_TEACHER
+    end
+  end
+
+  describe '#verified_teacher?' do
+    context 'when the user is a normal teacher' do
+      let(:teacher) {create :teacher}
+      it 'returns false' do
+        teacher.valid?
+        _(teacher.verified_teacher?).must_equal false
+      end
+    end
+    context 'when the user is has verified permissions' do
+      let(:teacher) {create :teacher}
+      before do
+        teacher.permission = UserPermission::AUTHORIZED_TEACHER
+      end
+      it 'returns true' do
+        teacher.valid?
+        _(teacher.verified_teacher?).must_equal true
+      end
+    end
+    context 'when the user is in a plc course' do
+      let(:teacher) {create :teacher}
+      before do
+        create(:plc_user_course_enrollment, user: teacher, plc_course: create(:plc_course))
+      end
+      it 'returns true' do
+        teacher.valid?
+        _(teacher.verified_teacher?).must_equal true
+      end
+    end
+    context 'when the user is a student' do
+      it 'returns false' do
+        student = create :student
+        student.valid?
+        _(student.verified_teacher?).must_equal false
+      end
+    end
+  end
+  describe '#verified_instructor?' do
+    context 'when the user is a normal teacher' do
+      let(:teacher) {create :teacher}
+      it 'returns false' do
+        teacher.valid?
+        _(teacher.verified_instructor?).must_equal false
+      end
+    end
```

Comment:
I would use a subject block since we are testing the same method in all assertions.
```suggestion
  describe '#verified_instructor?' do
    subject(:verified_instructor?) {user.verified_instructor?}
    context 'when the user is a normal teacher' do
      let(:user) {create :teacher}
      
      it {_verified_instructor?.must_equal false}
    end
```
Also, when the test setup performs one assertion, you can condense your test into one line, making things way more readable and shortening your tests.

Reply:
Good call, refactored to use the subject block.